### PR TITLE
docs(readme): update

### PR DESCRIPTION
**Pull request by teto**
see commit messages but to sum up:
- moved down ``Cargo-like rocks.toml file for declaring all your plugins` since I find this a weak feature, less powerful for sure than what was coming next aka "automatic dependency management"
- Added a section to mention the competition

I dont like that we mention twice the ecosystem about all the extensions but not sure how to do better so I left it untouched.
I do think that section should appear in `:h rocks.nvim` (currently is the barebone `A luarocks plugin manager for Neovim.` xD) because users are bound to want to install plugins from git so at least rocks-git.nvim should appear in help IMO.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -498,6 +500,13 @@ which contains the `luarocks` stderr output, among other logs.
 - [luarocks.nvim](https://github.com/vhyrro/luarocks.nvim):
   Adds basic support for installing lua rocks to [lazy.nvim](https://github.com/folke/lazy.nvim)
 
+
+## Other neovim plugin managers
```
Comment (mrcjkb) [inline README.md:504]:
```suggestion
## :link: Other neovim plugin managers
```

File: README.md [hunk 2]
```diff
@@ -498,6 +500,13 @@ which contains the `luarocks` stderr output, among other logs.
...
+
+- [lazy.nvim](https://github.com/folke/lazy.nvim): started luarocks support in
+  version [11.X](https://lazy.folke.io/news#11x)
+- [vim-plug](https://github.com/junegunn/vim-plug)
+
```
Comment (mrcjkb) [inline README.md:509]: note: It feels a bit strange to me mentioning only lazy.nvim and vim-plug here.
We should either move the lazy.nvim mention to the above section.
Or we should mention additional plugin mangers.
Some that come to mind:

- [pckr](https://github.com/lewis6991/pckr.nvim)
- [paq-nvim](https://github.com/savq/paq-nvim)
- [mini.deps](https://github.com/echasnovski/mini.deps)

**Review by mrcjkb — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (teto) [reply README.md:509]: absolutely. Do you think I should merge this section with the previous ?

Comment (teto) [reply README.md:509]: I think I like the separation though because it makes it clear those are "competition"

Comment (teto) [reply README.md:509]: similar type of rocks.nvim. Could be a subsection of the previous one as well.

Comment (mrcjkb) [reply README.md:509]: I think two sections are fine.

**Review by mrcjkb — APPROVED** [review]

**Pull request merged**